internal/backup_fetch_handler_new.go

```diff
@@ -28,38 +83,41 @@ func GetPgFetcherNew(dbDataDirectory, fileMask, restoreSpecPath string) func(fol
 			tracelog.ErrorLogger.FatalfOnError("Failed to fetch backup: %v\n",
 				newNonEmptyDbDataDirectoryError(dbDataDirectory))
 		}
-
-		err = deltaFetchRecursionNew(backup.Name, folder, utility.ResolveSymlink(dbDataDirectory), spec, filesToUnwrap)
+		config := newFetchConfig(backup.Name, utility.ResolveSymlink(dbDataDirectory), folder, spec, filesToUnwrap)
+		err = deltaFetchRecursionNew(config)
 		tracelog.ErrorLogger.FatalfOnError("Failed to fetch backup: %v\n", err)
 	}
 }
 
 // TODO : unit tests
 // deltaFetchRecursion function composes Backup object and recursively searches for necessary base backup
-func deltaFetchRecursionNew(backupName string, folder storage.Folder, dbDataDirectory string,
-	tablespaceSpec *TablespaceSpec, filesToUnwrap map[string]bool) error {
-	backup, err := GetBackupByName(backupName, utility.BaseBackupPath, folder)
+func deltaFetchRecursionNew(cfg *FetchConfig) error {
+	backup, err := GetBackupByName(cfg.backupName, utility.BaseBackupPath, cfg.folder)
 	if err != nil {
 		return err
 	}
 	sentinelDto, err := backup.GetSentinel()
 	if err != nil {
 		return err
 	}
-	chooseTablespaceSpecification(sentinelDto, tablespaceSpec)
+	chooseTablespaceSpecification(sentinelDto, cfg.tablespaceSpec)
 
 	if sentinelDto.IsIncremental() {
-		tracelog.InfoLogger.Printf("Delta %v at LSN %x \n", backupName, *(sentinelDto.BackupStartLSN))
-		baseFilesToUnwrap, err := GetBaseFilesToUnwrap(sentinelDto.Files, filesToUnwrap)
+		tracelog.InfoLogger.Printf("Delta %v at LSN %x \n", cfg.backupName, *(sentinelDto.BackupStartLSN))
+		baseFilesToUnwrap, err := GetBaseFilesToUnwrap(sentinelDto.Files, cfg.filesToUnwrap)
 		if err != nil {
 			return err
 		}
 		unwrapResult, err := backup.unwrapNew(cfg.dbDataDirectory, sentinelDto, cfg.filesToUnwrap, false)
```

Comment:
unwrapResult is not used in the previous commit. It's okay because PR works, and I need several commits only for readability) But FYI it's a good practice to have ability to revert to any commit and get code that still works.